File: cs/Minsk/CodeAnalysis/Binding/BoundLiteralExpression.cs
namespace Minsk.CodeAnalysis.Binding;

internal sealed class BoundLiteralExpression : BoundExpression
{
    public BoundLiteralExpression(object value)
    {
        Value = value;
    }

    public object Value { get; }
    public override Type Type => Value.GetType();
    public override BoundNodeKind Kind => BoundNodeKind.LiteralExpression;
}

File: kt/Minsk/minsk/src/main/kotlin/dev/phytolizer/minsk/analysis/binding/BoundLiteralExpression.kt
package dev.phytolizer.minsk.analysis.binding

import java.lang.reflect.Type

internal class BoundLiteralExpression(val value: Any) : BoundExpression() {
    override val type: Type
        get() = value.javaClass
    override val kind: BoundNodeKind
        get() = BoundNodeKind.LiteralExpression
}
